### PR TITLE
docs: replace `@nuxt/kit` example with node built-ins

### PR DESCRIPTION
**Pull request by warflash**
### 🔗 Linked issue

#19852 

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Currently the docs recommned using @nuxt/kit for path resolution which does not work.
To prevent confusion this PR adds a working path resolution example using node built-ins.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/2.guide/3.going-further/7.layers.md [hunk 1]
```diff
@@ -160,13 +160,14 @@ When importing using aliases (such as `~/` and `@/`) in a layer components and c
 Also when using relative paths in `nuxt.config` file of a layer, (with exception of nested `extends`) they are resolved relative to user's project instead of the layer. As a workaround, use full resolved paths in `nuxt.config`:
 
 ```js [nuxt.config.ts]
-import { createResolver } from '@nuxt/kit'
+import { fileURLToPath } from 'url'
+import { dirname, join } from 'path'
 
-const { resolve } = createResolver(import.meta.url)
+const __dirname = dirname(fileURLToPath(import.meta.url))
```
Comment (danielroe) [inline docs/2.guide/3.going-further/7.layers.md:166]: Might not matter here, but I would advise avoiding `__dirname` as it's sometimes used to detect a CJS environment.

Comment (pi0) [reply docs/2.guide/3.going-further/7.layers.md:166]: I agree that we might encourage all ESM syntax with `impoort.meta.url`

BTW jiti probably will always support CJS (`require.resolve` is even available in `nuxt.config` and works perfectly fine...) it is only and only matter of consistency to not use this simpler syntax..

Comment (danielroe) [reply docs/2.guide/3.going-further/7.layers.md:166]: I mean, avoid the variable name `__dirname` as a matter of good practice.

Comment (pi0) [reply docs/2.guide/3.going-further/7.layers.md:166]: I understand and it is a nice practice to stick with ESM syntax.

Only mentioning it _is_ a cjs environment by jiti for loading config. there should be no issues with it.

**Review by danielroe — APPROVED** [review]

**Pull request merged**